pages/Stats.vue

```diff
@@ -0,0 +1,282 @@
+<template>
+  <div
+    id="stats-viewer"
+    class="mx-4 mt-4 flex h-[91vh] flex-col"
+  >
+    <div class="stats-header">
+      <h1 class="text-2xl font-bold text-gray-700">
+        Statistics
+      </h1>
+    </div>
+    <div class="flex items-center justify-between">
+      &nbsp;
+      <div class="font-semibold text-center">Total Examined Names: {{ numRecords }}</div>
+      <div class="flex items-center space-x-2">
+        <div class="checkbox-wrapper">
+          <input
+            id="stats-checkbox"
+            v-model="myStats"
+            type="checkbox"
+          >
+          &nbsp;<label for="stats-checkbox" class="font-bold" >My Stats</label>
+        </div>
+        <div class="timespan-input  align-baseline">
+          <label
+            for="timespan"
+            class="font-bold align-baseline"
+          >Hours:&nbsp;</label>
+          <TextInput
+            id="timespan"
+            ref="numberinput"
+            v-model="timespan"
+            type="number"
+            style="width: 100px; margin-right: 10px;"
+          />
+          <button
+            class="bg-blue-800 text-white font-bold py-2 px-4 rounded"
+            @click="fetchStats"
+          >
+            Get Stats
+          </button>
```

Comment:
Use the `IconButton` component
```suggestion
          <IconButton @click="fetchStats">
            Get Stats
          </IconButton>
```